### PR TITLE
Decorate a class code

### PR DESCRIPTION
**Pull request by chepkok3**
- [ ] Interface
- Created a class Nameable by implementing a method called correct_name that will raise a NotImplementedError.
- [ ] Turned the Person class to Nameable by:
- Making sure that the Person class inherits from Nameable
- Making sure that the class has a method correct_name implemented. It should simply return the name attribute.
- [ ] Prepare base Decorator by:
- Making sure that it inherits from Nameable.
- Assigning a nameable object from params to an instance variable in the constructor.
- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- [ ] Prepared CapitalizeDecorator and TrimmerDecorator 
- [ ] For the CapitalizeDecorator:
- Created a class that inherits from the base Decorator class.
- Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
- [ ] For the TrimmerDecorator:
- Created a class that inherits from the base Decorator class.
- Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +76,7 @@ Example command:
 
 Install this project with:
 
-- cd opp-school-library
+- cd oop-school-library
```
Comment (tufoinnkuo10) [inline README.md]: Nice job so far. let's make some final touches to make everything perfect
- [ ] your issue link under the contributing section in the README is broken. Please kindly add the correct link to make it professional
- [ ] the Email link you added under the author section is broken as well. Please add a correct one

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @chepkok3 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ Name well trimed


Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: README.md (hunk 1 above)
Comment (chepkok3) [reply README.md]: @tufoinnkuo10 Sure. Let me do that

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @chepkok3 👍, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
 Use the Decorator design pattern ✔️ 
Create a class that inherits from the base Decorator class ✔️ 

 ## Required Changes ♻️

- [ ] So sorry about this, I still got 404 error when I clicked on your issue link in the README file. I guess you did not understand very well what the previous reviewer requested. 

Untitled

The correct link suppose to be `https:/github.com/chepkok3/OOP-school-library/issues` and not `https://github.com/chepkok3/OOP-school-library/https:/github.com/chepkok3/OOP-school-library/issues`.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @chepkok3, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**